### PR TITLE
Unify Analysis Options YAML into a single reference file (see #74)

### PR DESCRIPTION
**Pull request by miquelbeltran**
This PR unifies all projects `analysis_options.yaml` into a single one and upgrades pedantic to 1.9.2, fixing all new analysis issues raised.

### CODE REVIEW

**Comment by miquelbeltran** [comment]
It's a massive PR, sorry, not a lot of lines changed but a lot of files.

**Comment by jpnurmi** [comment]
> It's a massive PR, sorry, not a lot of lines changed but a lot of files.

Could be done in steps one package at a time to make it easier to review? :)

**Comment by miquelbeltran** [comment]
Maybe yes, in this case is because updating the version of `pedantic` on melos so it affected all packages. Some packages were at `1.8.0` while some others at `1.9.0`.

**Comment by jpnurmi** [comment]
Is the relative include from the repo root necessary? Would that break stand-alone packages downloaded from pub.dev?

P.S. Do you know how `flutter/plugins` does it? They seem to have one `analysis_options.yaml` in the repo root without any additional includes in the packages:
```
flutter/plugins (master)$ git grep analysis_options
analysis_options.yaml:include: package:pedantic/analysis_options.1.8.0.yaml
packages/video_player/video_player_web/CHANGELOG.md:* Add `analysis_options.yaml` to the package, so we can ignore `undefined_prefixed_name` errors. Works around https://github.com/flutter/flutter/issues/41563.
script/incremental_build.sh:# Plugins that deliberately use their own analysis_options.yaml.
```

**Comment by miquelbeltran** [comment]
you are right, it is not necessary to have the file with the include except if you are specifying different rules, so I remove them all and just leave the root one

File: packages/battery_plus/lib/battery_plus.dart [hunk 1]
```diff
@@ -23,9 +23,7 @@ class Battery {
   // EventChannel because it is overridden. Forcing the class to be a singleton class can prevent
   // misuse of creating a second instance from a programmer.
   factory Battery() {
-    if (_singleton == null) {
-      _singleton = Battery._();
-    }
+    _singleton ??= Battery._();
```
Comment (jpnurmi) [inline packages/battery_plus/lib/battery_plus.dart:26]: :+1: Nice, I've been tempted to fix these

File: packages/package_info_plus/example/analysis_options.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+include: ../../../analysis_options.yaml
```
Comment (jpnurmi) [inline packages/package_info_plus/example/analysis_options.yaml:1]: can be removed?

File: packages/package_info_plus_macos/pubspec.yaml [hunk 3]
```diff
@@ -18,6 +18,3 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-
-dev_dependencies:
-  pedantic: ^1.8.0
```
Comment (jpnurmi) [inline packages/package_info_plus_macos/pubspec.yaml:23]: is this correct or should it be bumped to 1.9.2?

File: packages/package_info_plus_web/lib/package_info_plus_web.dart [hunk 4]
```diff
@@ -17,8 +17,8 @@ class PackageInfoPlugin extends PackageInfoPlatform {
 
   @override
   Future<PackageInfoData> getAll() async {
-    String url =
-        "${Uri.parse(window.document.baseUri).removeFragment()}/version.json";
+    var url =
```
Comment (jpnurmi) [inline packages/package_info_plus_web/lib/package_info_plus_web.dart:20]: final?

File: packages/sensors_plus/example/lib/snake.dart [hunk 5]
```diff
@@ -33,17 +33,17 @@ class SnakeBoardPainter extends CustomPainter {
...
       ..color = Colors.black
       ..style = PaintingStyle.fill;
     canvas.drawRect(
       Rect.fromPoints(Offset.zero, size.bottomLeft(Offset.zero)),
       blackLine,
     );
-    for (math.Point<int> p in state.body) {
-      final Offset a = Offset(cellSize * p.x, cellSize * p.y);
-      final Offset b = Offset(cellSize * (p.x + 1), cellSize * (p.y + 1));
+    for (var p in state.body) {
```
Comment (jpnurmi) [inline packages/sensors_plus/example/lib/snake.dart:44]: final?

**Review by jpnurmi — APPROVED** [review]
It's a big PR but the changes are very simple. I glanced quickly through and have only some nit-pick. Fix if you feel like it. :)

File: packages/package_info_plus_macos/pubspec.yaml (hunk 3 above)
Comment (miquelbeltran) [reply packages/package_info_plus_macos/pubspec.yaml:23]: well spotted, actually there's not even dart code on the macos projects so I think the dependency could be removed

**Comment by miquelbeltran** [comment]
I am not sure why I got some analysis issues on my computer by they don't appear on CI, I created a ticket for that: https://github.com/fluttercommunity/plus_plugins/issues/85

Besides, I am merging this now :)

**Comment by miquelbeltran** [comment]
Thanks for reviewing @jpnurmi !

**Pull request merged**